### PR TITLE
Move to Travis CI for multiple Rust versions

### PR DESCRIPTION
**Pull request by marmistrz**

### CODE REVIEW

**Comment by kubkon** [comment]
Perhaps you could consider Azure Pipelines? I've recently started migrating from Travis to Azure, and it's not bad especially if you're thinking of targeting all 3 major hosts: linux, macos and windows. Anyhow, here's a great write-up of how to get up and running with Rust on Azure: [Azure Pipelines for Rust projects](https://nbsoftsolutions.com/blog/azure-pipelines-for-rust-projects), and [here](https://github.com/golemfactory/sp-wasm/blob/master/azure-pipelines.yml) you can find my config for one of my projects which I've successfully migrated this very morning :grin:

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+language: rust
+rust:
+  - 1.33.0
+  - stable
+  - beta
+  - nightly
+cache: cargo
```
Comment (kubkon) [inline .travis.yml]: Caching `cargo` in Travis is known to be error prone and not necessarily speeding the builds up in an expected way. [Here](https://levans.fr/rust_travis_cache.html)'s a great write-up on the topic with a suggested fix.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+before_script:
+  - rustup component add rustfmt clippy
+script:
+  - cargo build
+  - cargo test
+  - cargo fmt -- --check
```
Comment (kubkon) [inline .travis.yml]: For the future, perhaps `cargo fmt --all -- --check` would be better?

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  - 1.33.0
+  - stable
+  - beta
+  - nightly
+cache: cargo
+before_script:
+  - rustup component add rustfmt clippy
+script:
+  - cargo build
+  - cargo test
```
Comment (kubkon) [inline .travis.yml]: For the future, perhaps `cargo test --all` would be better?

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,14 @@
+language: rust
+rust:
+  - 1.33.0
+  - stable
+  - beta
+  - nightly
+cache: cargo
+before_script:
+  - rustup component add rustfmt clippy
```
Comment (kubkon) [inline .travis.yml]: Here's I'd strongly advise to be explicit in toolchain choice:
```
rustup +stable component add rustfmt clippy
```

**Comment by marmistrz** [comment]
@kubkon thanks for the review. I wasn't aware that Mozilla had so many issues with travis. For reference, here's the list of the issues that Mozilla had with travis and why they decided to move away from it. https://internals.rust-lang.org/t/which-ci-platform-should-rust-use/9322

Still, I think that for now I'll use Travis anyway, because it's much more work to migrate to Azure Pipelines and I hope that with rust-lang migrating to Azure Pipelines out-of-the-box Rust support should be available in the future.

**Pull request merged**